### PR TITLE
display entity selector inline in menu

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

This change due to a conversion with an advanced user who didn't know about the entity selector.
I change the current UX to let the content to be displayed as a sub dropdown (and be more predictable)

image

### CODE REVIEW

**Comment by orthagh** [comment]
I will remove today the ajax loading by a direct integration

**Comment by orthagh** [comment]
Done

**Review by trasher — APPROVED** [review]

**Comment by orthagh** [comment]
To check also, we could encounter performance issue as the entity tree will always be loaded.
I'm inclined to remove the auto selection of current entity in the tree as when you need the component you will do a new search most of the time

File: templates/layout/parts/profile_selector.html.twig [hunk 1]
```diff
@@ -16,19 +17,116 @@
...
+               <i class="fas fa-search"></i>
+            </button>
+            <a class="btn btn-icon btn-outline-secondary" href="#" id="entsearchtext{{ rand }}_clear"
+               title="{{ __("Clear search") }}" data-bs-toggle="tooltip" data-bs-placement="top">
+                  <i class="fas fa-times"></i>
+            </a>
+            <a href="{{ target }}?active_entity=all" class="btn btn-secondary"
+               title="{{ __('Select all') }}" data-bs-toggle="tooltip" data-bs-placement="top">
+               <i class="fas fa-eye"></i>
+            </a>
```
Comment (cedric-anne) [inline templates/layout/parts/profile_selector.html.twig:56]: At first glance, I did not understand what this "eye" could be used for.

Is this "Select all" feature really usefull, as long as is is accessible via the icon located next to the root entity ?

File: templates/layout/parts/profile_selector.html.twig [hunk 2]
```diff
@@ -16,19 +17,116 @@
...
 <script type="text/javascript">
 $(function() {
-   $('.show_entity_modal').click(function() {
-      glpi_ajax_dialog({
-         url: '{{ path('/ajax/entitytree.php') }}',
-         title: __('Select the desired entity'),
-         id: 'entity_modal',
-      });
+   $('#tree_projectcategory{{ rand }}').jstree({
+      'plugins': ['search', 'qload', 'conditionalselect'],
```
Comment (cedric-anne) [inline templates/layout/parts/profile_selector.html.twig:67]:
```suggestion
      'plugins': ['search', 'conditionalselect'],
```

File: templates/layout/parts/profile_selector.html.twig [hunk 3]
```diff
@@ -16,19 +17,116 @@
...
+         'ajax': {
+            'type': 'POST',
+            'url': '{{ path("/ajax/entitytreesearch.php") }}'
+         }
+      },
+      'qload': {
+         'prevLimit': 50,
+         'nextLimit': 30,
+         'moreText': '{{ __("Load more entities...") }}'
+      },
```
Comment (cedric-anne) [inline templates/layout/parts/profile_selector.html.twig:80]:
```suggestion
```

File: templates/layout/parts/profile_selector.html.twig [hunk 4]
```diff
@@ -16,19 +17,116 @@
...
+      {{ session('glpiactive_entity_name')|verbatim_value|u.truncate(35, '...') }}
+   </a>
+   <div class="dropdown-menu p-3">
+      <h3>{{ __('Select the desired entity') }}</h3>
+
+      <div class="alert alert-info" role="alert">
+         <i class="fas fa-info-circle"></i>
+         <span class="ms-2">
+            <img src="{{ path('pics/entity_all.png') }}" alt="entity_all">
+            {{ __('to see the entity and its sub-entities') }}
```
[the remark this answers is not in the document]
Comment (cedric-anne) [reply templates/layout/parts/profile_selector.html.twig:37]: Here is a preview, maybe some CSS rules can be added.

Capture d’écran du 2021-09-23 14-24-45

[the remark this answers is not in the document]
Comment (orthagh) [reply templates/layout/parts/profile_selector.html.twig:37]: This part will be managed in #9596

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**